extension/entitystore/serviceprovider.go

```diff
@@ -257,9 +264,27 @@ func (s *serviceprovider) scrapeImdsServiceName() error {
 			break
 		}
 	}
+	if strings.Contains(tags, ec2tagger.Ec2InstanceTagKeyASG) {
+		asg, err := s.metadataProvider.InstanceTagValue(context.Background(), ec2tagger.Ec2InstanceTagKeyASG)
+		if err != nil {
+			s.logger.Error("Failed to get AutoScalingGroup through metadata provider", zap.Error(err))
```

Comment:
nit: might be good to mention this does not affect agent functionality

Reply:
actually I think we should change this debug log line since this is not affecting agent functionality but if user does not have IMDS tag enabled, they will just see this error log being spammed over and over again

Reply:
Makes sense will update in revision